### PR TITLE
Feature/implement reports api

### PR DESCRIPTION
**Pull request by andrewperis**
Implemented the Details Report and inserted placeholders for the Summary Report and Weekly Report.

### CODE REVIEW

File: Clockify.Net/Models/Reports/DetailedReportDto.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+using Clockify.Net.Models.HourlyRates;
+using Clockify.Net.Models.Memberships;
+
+namespace Clockify.Net.Models.Reports
+{
+    public class DetailedReportDto
+    {
+        public List<TotalsDto> Totals { get; set; }
+        public List<TimeEntryDto> Timeentries { get; set; }
```
Comment (Morasiu) [inline Clockify.Net/Models/Reports/DetailedReportDto.cs:10]: Shouldn't this be `TimeEntries`?

File: Clockify.Tests/Tests/ReportTests.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    Page = 1,
+                    PageSize = 50
+                },
+                TimeZone = userResponse.Data.Settings.TimeZone
+            };
+
+            var getDetailedReportResult = await _client.GetDetailedReportAsync(_workspaceId, detailedReportRequest);
+            getDetailedReportResult.IsSuccessful.Should().BeTrue();
+            getDetailedReportResult.Data.Should().NotBeNull();
+            getDetailedReportResult.Data.Timeentries.Count.Should().BeGreaterOrEqualTo(1);
```
Comment (Morasiu) [inline Clockify.Tests/Tests/ReportTests.cs:115]: I think that FluentAssertion have method with syntax `getDetailedReportResult.Data.Timeentries.HaveCountSomethingSomething()`. That would be better if possible. If not let's leave it.

**Review by Morasiu — CHANGES_REQUESTED** [review]

**Comment by Morasiu** [comment]
Great work. Clean and even have some tests!

One question, will the Summary Report and Weekly Report be completed in this PR or not? Just asking if I can merge it later in the current form.

**Review by andrewperis — COMMENTED** [review]
Thanks!

I was not planning on including the Summary Report and Weekly Report in this PR. It'll be a separate effort.

File: Clockify.Tests/Tests/ReportTests.cs (hunk 2 above)
Comment (Morasiu) [reply Clockify.Tests/Tests/ReportTests.cs]: `getDetailedReportResult.Data.Timeentries.Should().HaveCountGreaterOrEqualTo(1)` -< That one should work. Let me know if you can see it.

Comment (andrewperis) [reply Clockify.Tests/Tests/ReportTests.cs]: Yes, I can see that now. Not sure why I couldn't find it yesterday. Thanks for pointing it out!

**Review by Morasiu — APPROVED** [review]

**Pull request merged**